src/codegen/codegen_c_visitor.cpp

```diff
@@ -133,7 +133,9 @@ void CodegenCVisitor::visit_var_name(const VarName& node) {
     }
     if (index) {
         printer->add_text("[");
+        printer->add_text("((int)");
```

Comment:
```suggestion
        printer->add_text("static_cast<int>(");
```
[ES.49: If you must use a cast, use a named cast](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-casts-named).